flask_bcrypt.py

```diff
@@ -155,8 +155,19 @@ def generate_password_hash(self, password, rounds=None):
             password = password.encode('u8')
         elif PYVER >= 3 and isinstance(password, bytes):
             password = password.decode('utf-8')
-        password = str(password)
-        return bcrypt.hashpw(password, bcrypt.gensalt(rounds))
+
+        if PYVER < 3:
+            password = str(password)
```

Comment:
This will raise `UnicodeDecodeError` in python2 if `password` is a unicode string that contains non-ascii characters.